### PR TITLE
Tags Model & Creation | Improve Authorization Utils and Tests | Add Groups in User Controller Responses

### PR DESCRIPTION
**Pull request by thudsonbu**
## Description

[Proposal](https://github.com/SocexSolutions/agenda-v2/issues/318)

This pr adds support for creating tags as well as various refactorings that improve the quality of our tests.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Housekeeping

## How Has This Been Tested?

Full coverage unit tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/lib/models/tag.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const tag = new mongoose.Schema({
+  group_id: {
+    type: Schema.Types.ObjectId,
+    ref: "User",
```
Comment (zbarnz) [inline api/lib/models/tag.js:7]: Group

File: api/test/fakes/register.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+let count = 0;
+
+module.exports = (overrides) => {
+  count += 1;
```
Comment (zbarnz) [inline api/test/fakes/register.js:4]: I dont understand the count also isnt the definition outside of scope?

File: api/test/tests/middleware/authenticate.test.js [hunk 3]
```diff
@@ -38,10 +39,11 @@ describe(modulePath, () => {
     this.module = libRewire(modulePath);
   });
 
-  it("should accept an valid auth token", async () => {
+  it("should accept valid user token", async () => {
```
Comment (zbarnz) [inline api/test/tests/middleware/authenticate.test.js:42]:
```suggestion
  it("should accept a valid user token", async () => {
```

**Review by zbarnz — CHANGES_REQUESTED** [review]
Some very important changes are required

File: api/test/fakes/register.js (hunk 2 above)
Comment (thudsonbu) [reply api/test/fakes/register.js:4]: Its in the module scope which is available to the function. It's a really easy way to make sure that we don't have any conflicts.

**Review by zbarnz — APPROVED** [review]

**Pull request merged**